lib/src/message/trailer.rs

```diff
@@ -0,0 +1,211 @@
+///
+enum TrailerCollector<'l> {
+    Collecting(&'l mut LinkedList<Trailer>),
+    Dumping,
+}
+
+impl<'l> TrailerCollector<'l> {
+    pub fn new(target: &'l mut LinkedList<Trailer>) -> Self {
+        TrailerCollector::Collecting(target)
+    }
+
+    /// Dump the current and all future trailers pushed to this collector
+    ///
+    pub fn dumping(self) -> Self {
+        if let TrailerCollector::Collecting(target) = self {
+            target.clear();
+        }
+        TrailerCollector::Dumping
+    }
+
+    /// Push a new trailer
+    ///
+    /// The trailer pushed will be either collected or dumped, based on the
+    /// current state.
+    ///
+    pub fn push(self, trailer: Trailer) -> Self {
+        match self {
+            TrailerCollector::Collecting(target) => {
+                target.push_back(trailer);
+                TrailerCollector::Collecting(target)
+            },
+            TrailerCollector::Dumping => self,
+        }
+    }
+}
+
+
+pub struct Trailers<'a> {
+    lines: Lines<'a>,
+    buf: LinkedList<Trailer>,
+}
+
+impl<'a> Trailers<'a> {
+
+    /// Create a new Trailers iterator from a commit message
+    pub fn new(text: &'a str) -> Trailers<'a> {
+        Trailers {
+            lines: Lines::new(text),
+            buf: LinkedList::new(),
+        }
+    }
+
+    pub fn only_dit(self) -> DitTrailers<'a> {
+        DitTrailers(self)
+    }
+
+}
+
+impl<'a> Iterator for Trailers<'a> {
+    type Item = Trailer;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        'outer: loop {
+            if let Some(trailer) = self.buf.pop_front() {
+                return Some(trailer);
+            }
+
+            // refill buffer from next block
+            let mut collector = TrailerCollector::new(&mut self.buf);
+            let mut at_end = true;
+
+            'refill: for line in self.lines.next() {
+                at_end = false;
+                collector = match line {
+                    Line::Text(_) => collector.dumping(), // block of text
+                    Line::Trailer(t) => collector.push(t),
+                    Line::Blank => continue 'outer, // end of block
+                }
+            }
+
+            if at_end {
+                return None;
+            }
+        }
+    }
+
+}
+
+pub struct DitTrailers<'a>(Trailers<'a>);
+
+impl<'a> Iterator for DitTrailers<'a> {
+    type Item = Trailer;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        loop {
+            match self.0.next() {
+                None => return None,
+                Some(trailer) => {
+                    if trailer.key.0.starts_with("Dit") {
+                        return Some(trailer);
+                    } else {
+                        continue;
+                    }
+
+                }
+            }
+        }
+    }
+
+}
+
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+
+pub trait IntoMap {
+    fn into_hashmap(self) -> HashMap<TrailerKey, TrailerValue>;
```

Comment:
To be more specific, this part of the implementation is here: https://doc.rust-lang.org/stable/std/iter/trait.FromIterator.html